ddb_cp.module

```diff
@@ -125,6 +125,8 @@ function ddb_cp_get_test_result($site = NULL, $number = NULL) {
   else {
     return MENU_NOT_FOUND;
   }
+
+  // @TODO: This function never returns any XML data only page not found error code?
```

Comment:
If you follow the function call to _ddb_cp_save_test_result() you'll find that under normal circumstances the call ends in a call to drupal_exit(), finishing off the HTTP request. Otherwise, it'll dispatch an error message using drupal_set_message().

Have you actually tried the code?

It's intended to be called from a link in the control panel's "Last Test Result" fieldset that has an icon and a link to show the raw Jenkins XML testresult as a file. The code will trigger the browser's "Save file as..." functionality.